src/PaletteStore.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/inclusive-design/adaptive-palette/blob/main/LICENSE
+ */
+
+"use strict";
+
+export class PaletteStore {
+
+  // Singleton storage for all palettes
+  // The contents are named Palette instances; hence, each palette must have
+  // a unique name.
+  static paletteMap = {};
+
+  /**
+   * Report if the PaletteStore is empty.
+   * @return: `true` if the store is empty; `false` otherwise.
+   */
+  isEmpty () {
+    return Object.keys(PaletteStore.paletteMap).length === 0;
+  }
+
+  /**
+   * Add a palette to the store, or replace a palette with a new one.  If the
+   * palette's name/identifier matches a palette already in the store, it
+   * replaces it.
+   * @param: {Palette} palette - The palette to add to the store.
+   * @param: {Palette}.name  - The internal name of the palette.
+   * @param: {String} name   - Optional, the preferred name of the palette.
+   */
+  addPalette (palette, paletteName?: string) {
+    if (!palette) {
+      return;
+    }
+    let palName;
+    if (paletteName) {
+      palName = paletteName;
+    } else if (palette.name) {
+      palName = palette.name;
+    } else {
+      return;
+    }
+    PaletteStore.paletteMap[palName] = palette;
+    console.log(`Palette ${palName} added to the store.`);
+  }
+
+  /**
+   * Remove the palette with the given name.
+   * @param: {String} paletteName - The palette to remove.
+   * @return {Palette} reference to the removed palette.
+   */
+  removePalette (paletteName) {
```

Comment:
Add a type check for `paletteName`. Same comment for line 93.

Reply:
Yes, it is a `string` type.